### PR TITLE
Circleci setup

### PR DESCRIPTION
**Pull request by vijtrip2**
Issue #, if available:

Description of changes:
* Added the file for circleci setup
* Added run-integration-tests.sh file and helper files copied from amazon-vpc-cni-k8s package. Most of long running tasks are commented out.
* The integration test validates the circleci setup and aws cli setup.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/lib/cluster.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        AWS_K8S_TESTER_EKS_S3_BUCKET_NAME=$S3_BUCKET_NAME \
+        AWS_K8S_TESTER_EKS_S3_BUCKET_CREATE=$S3_BUCKET_CREATE \
+        AWS_K8S_TESTER_EKS_PARAMETERS_VERSION=${K8S_VERSION%.*} \
+        AWS_K8S_TESTER_EKS_PARAMETERS_ENCRYPTION_CMK_CREATE=false \
+        AWS_K8S_TESTER_EKS_PARAMETERS_ROLE_CREATE=$ROLE_CREATE \
+        AWS_K8S_TESTER_EKS_PARAMETERS_ROLE_ARN=$ROLE_ARN \
+        AWS_K8S_TESTER_EKS_ADD_ON_MANAGED_NODE_GROUPS_ENABLE=true \
+        AWS_K8S_TESTER_EKS_ADD_ON_MANAGED_NODE_GROUPS_ROLE_CREATE=$ROLE_CREATE \
+        AWS_K8S_TESTER_EKS_ADD_ON_MANAGED_NODE_GROUPS_ROLE_ARN=$ROLE_ARN \
+        AWS_K8S_TESTER_EKS_ADD_ON_MANAGED_NODE_GROUPS_MNGS='{"GetRef.Name-mng-for-cni":{"name":"GetRef.Name-mng-for-cni","remote-access-user-name":"ec2-user","tags":{"group":"amazon-vpc-cni-k8s"},"release-version":"","ami-type":"AL2_x86_64","asg-min-size":3,"asg-max-size":3,"asg-desired-capacity":3,"instance-types":["c5.xlarge"],"volume-size":40}}' \
```
Comment (varun1524) [inline scripts/lib/cluster.sh:28]: NIT: amazon-service-operator-k8s in place of amazon-vpc-cni-k8s and ack in place of cnis.

Comment (vijtrip2) [reply scripts/lib/cluster.sh:28]: copy-paste error. Thanks!

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+references:
+  environment: &env
+    GO111MODULE: "on"
+  filters: &branchfilters
+    branches:
+      only: mvp
+
+jobs:
+  build:
```
Comment (A-Hilaly) [inline .circleci/config.yml:18]: Since this is step is only running unit tests, maybe we can name it `test` or `unit_tests` ?

**Comment by mhausenblas** [comment]
Good call @A-Hilaly and also seems it's ATM failing?

**Comment by A-Hilaly** [comment]
@mhausenblas The [errors](https://app.circleci.com/pipelines/github/aws/aws-service-operator-k8s/2/workflows/4958b311-3047-4fe8-9b90-bf53a6f9eb03/jobs/2/steps) appearing after #52 merge are normal because the CircleCI configuration file was missing (solved in this PR).

Unit tests are passing after #53 merge so all good 👍 . There is just a [hold](https://app.circleci.com/pipelines/github/aws/aws-service-operator-k8s/3/workflows/c1e9e07e-733d-4123-846e-858e63beceab) waiting for a manual approval to trigger integration tests.

File: .circleci/config.yml (hunk 2 above)
Comment (jaypipes) [reply .circleci/config.yml:18]: Unfortunately, CircleCI requires a job called "build". :( Yeah, weird, but true..

Comment (mhausenblas) [reply .circleci/config.yml:18]: Is there a reason we're using Circle? Why not GH Actions?

Comment (A-Hilaly) [reply .circleci/config.yml:18]: @jaypipes it is true that CircleCI requires a job called "build" if the `workflows` section isn't defined.

I just tested this, it works fine:
```yaml
version: 2
jobs:
  test:
    docker: 
      - image: alpine
    steps:
      - checkout
      - run: echo "hello world"

workflows:
  version: 2
  test_only:
    jobs:
      - test
```

Comment (jaypipes) [reply .circleci/config.yml:18]: ah, nice catch. leftover knowledge from working with an earlier CircleCI setup, I guess!

Comment (jaypipes) [reply .circleci/config.yml:18]: > Is there a reason we're using Circle? Why not GH Actions?

The CNI and eks-charts repos use CircleCI, so we're copying the work I did for those repos. Also, there are some issues with permissions when using GH Actions. Ask @m00nf1sh for more details :)

Comment (vijtrip2) [reply .circleci/config.yml:18]: Thanks! https://github.com/aws/aws-service-operator-k8s/pull/54

**Pull request merged**